actionpack/lib/action_dispatch/middleware/flash.rb

```diff
@@ -45,6 +45,7 @@ module RequestMethods
       # read a notice you put there or <tt>flash["notice"] = "hello"</tt>
       # to put a new one.
       def flash
+        return Flash::NullFlash.new unless session.respond_to?(:loaded?)
```

Comment:
The object has no state, might as well have `NullFlash` be a self extending module as to not bother allocating a new one every time. Not that I expect this to be a hotspot, but it's easy enough to avoid.

Reply:
Good point. I addressed that and the comment about the signatures. Thanks again for reviewing!